hack/lib/util.sh

```diff
@@ -157,4 +164,45 @@ kube::util::gen-doc() {
   rm -rf "${tmpdir}"
 }
 
+# Takes a path $1 to traverse for md files to append the ga-beacon tracking
+# link to, if needed. If $2 is set, just print files that are missing
+# the link.
+kube::util::gen-analytics() {
+  local path="$1"
+  local dryrun="${2:-}"
+  local mdfiles dir link
+  # find has some strange inconsistencies between darwin/linux. The
+  # path to search must end in '/' for linux, but darwin will put an extra
+  # slash in results if there is a trailing '/'.
+  if [[ $( uname ) == 'Linux' ]]; then
+    dir="${path}/"
+  else
+    dir="${path}"
+  fi
+  # We don't touch files in Godeps|third_party, and the kubectl
+  # docs are autogenerated by gendocs.
+  mdfiles=($( find "${dir}" -name "*.md" -type f \
+              -not -path "${path}/Godeps/*" \
+              -not -path "${path}/third_party/*" \
+              -not -path "${path}/_output/*" \
+              -not -path "${path}/docs/kubectl*" ))
+  for f in "${mdfiles[@]}"
+  do
```

Comment:
nitpick: I think the usual syntax is to put the `do` or `then` on the same line as the command:

```shell
  for f in "${mdfiles[@]}"; do
```

Not sure how consistent this is in Kubernetes sources though...